src/sentry/notifications/utils/__init__.py

```diff
@@ -269,7 +269,10 @@ def has_integrations(organization: Organization, project: Project) -> bool:
 
 
 def is_alert_rule_integration(provider: IntegrationProvider) -> bool:
-    return any(feature == IntegrationFeatures.ALERT_RULE for feature in provider.features)
+    return any(
+        feature == (IntegrationFeatures.ALERT_RULE or IntegrationFeatures.ENTERPRISE_ALERT_RULE)
```

Comment:
fwiw this always evaluates as `feature == IntegrationFeatures.ALERT_RULE` -- mypy is now noticing that the latter half of the `or` is unreachable in the latest version (since `(IntegrationFeatures.ALERT_RULE or ...)` is the same as `IntegrationFeature.ALERT_RULE` because `IntegrationFeatures.ALERT_RULE` is truthy

it seems the intent is `feature in {IntegrationFeatures.ALERT_RULE, IntegrationFeatures.ENTERPRISE_ALERT_RULE}` ?

but actually there are no `IntegrationProvider`s that have the `ENTERPRISE_ALERT_RULE` feature (even the one edited in this PR -- as the edit is to the `metadata` but not the `IntegrationProvider.features` which still has normal `ALERT_RULE`) so for now I'm going to simplify the code to what's actually happening